src/hairpin.cpp

```diff
@@ -167,17 +169,31 @@ std::pair<int, int> Hairpin::GetBarlineOverlapAdjustment(int doubleUnit, int lef
     // default margin to consider them overlapping, which is adjusted in case we have wider barline on the left
     int leftAdjustment = 0;
     BarLine *leftBarline = startMeasure->GetLeftBarLine();
-    if (leftBarline) {
+    if (leftBarline && (spanningType == SPANNING_START_END || spanningType == SPANNING_START)) {
         int margin = doubleUnit;
         const int leftBarlineX = leftBarline->GetDrawingX();
         const int diff = leftX - leftBarlineX;
         if (leftBarline->GetForm() == BARRENDITION_rptstart) margin *= 1.5;
         if (diff < margin) leftAdjustment = margin - diff;
     }
-    // Same calculation is done for the right barline, with it having two barline forms that we need to consider
-    // as opposed to only one for the left barline
+
+    // Similar calculation is done for the right barline, with it having two barline forms that we need to consider
+    // as opposed to only one for the left barline. Additionally, when we have spanning hairpins, correct barline should
+    // be selected - when processing start of the spanning hairpin, we should check for the last measure of the current
+    // system, instead of the endMeasure
     int rightAdjustment = 0;
-    BarLine *rightBarline = endMeasure->GetRightBarLine();
+    BarLine *rightBarline = NULL;
+    if (spanningType == SPANNING_START_END || spanningType == SPANNING_END) {
+        rightBarline = endMeasure->GetRightBarLine();
+    }
+    else if (spanningType == SPANNING_START)
+    {
+        System *startSystem = vrv_cast<System *>(GetStart()->GetFirstAncestor(SYSTEM));
+        ClassIdComparison cmp(MEASURE);
```

Comment:
I usually `assert(startSystem)` even though I agree it is extremely unlikely to fail. Do you think it is too much? If you think so we would need to find a rule when to assert it and when not. But if we cannot define such a rule I would prefer simply always to assert them

Reply:
No, I agree that it's safer to check whether pointer is actually valid. However, I'd use a simple `if(...)` check instead of `assert(...)`, so that we do not crash right away just because of error in encoding or something among those lines.

Reply:
Glad to hear we are on the same line! Checking with `if` is good too. But for cases such as this one I think an `assert` is enough. I am fine either way.